pkg/flow/internal/controller/metrics.go

```diff
@@ -67,4 +68,7 @@ func (cc *controllerCollector) Collect(ch chan<- prometheus.Metric) {
 
 func (cc *controllerCollector) Describe(ch chan<- *prometheus.Desc) {
 	ch <- cc.runningComponentsTotal
+	for _, component := range cc.l.components {
+		component.register.Describe(ch)
+	}
```

Comment:
We shouldn't do this: Describe is only called once and will permanently register checked metrics. 

We should leave metrics from components as unchecked and remove this.

(Also, you needed to do cc.l.Components() to prevent a race condition)